examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -79,49 +101,293 @@ def best_low_complexity(cv_results):
     return best_idx
 
 
+# %%
+# Set up the pipeline and parameter grid
+# -------------------------------------
+#
+# We create a pipeline with two steps:
+# 1. Dimensionality reduction using PCA
+# 2. Classification using LinearSVC
+#
+# We'll search over different numbers of PCA components to find the optimal complexity.
+
 pipe = Pipeline(
     [
         ("reduce_dim", PCA(random_state=42)),
         ("classify", LinearSVC(random_state=42, C=0.01)),
     ]
 )
 
-param_grid = {"reduce_dim__n_components": [6, 8, 10, 12, 14]}
+param_grid = {"reduce_dim__n_components": [6, 8, 10, 12, 14, 16, 20, 25, 35, 45, 55]}
+
+# %%
+# Perform the search with GridSearchCV
+# -----------------------------------------
+#
+# We use GridSearchCV with our custom `best_low_complexity` function as the refit
+# parameter. This function will select the model with the fewest PCA components that
+# still performs within one standard deviation of the best model.
 
 grid = GridSearchCV(
     pipe,
     cv=10,
-    n_jobs=1,
+    n_jobs=1,  # increase this on your machine to use more physical cores
     param_grid=param_grid,
     scoring="accuracy",
```

Comment:
Maybe we could switch to `roc_auc` that is often more informative than `"accuracy"` (and not dependent on an arbitrary choice of decision threshold in particular).

Reply:
Actually, `roc_auc` would not work because it's only defined on binary classification problems.

Reply:
`roc_auc_ovr` is a bit of a hack for multi-class. Let's keep accuracy for the sake of a better multi-class metric.